### DISCUSSION TITLE
JRule and JSR223

### DISCUSSION THREAD

**Discussion by seaside1** [seaside1/jrule#108] [Ideas]
I have looked into this previously. It should be possible to make JRule compatible with JSR223.
However now looking into the core classes (RuleRegistry, ScriptedAutomationManager, SimpleRule and some others). 
I can see that it seems we would end up loosing a lot of control on how the rules are executed, and possibly also some of the conditions, which might miss some of the stuff we are using.

So my question is more if we think it's worth pursuing JSR223 compatibility or not? One advantage would be that you could use the Openhab GUI to view loaded and active rules, as well as making simple modifications and testing in openHAB GUI.

Personally I don't use the openHAB GUI and can live without it but would like to hear more opinions on the matter.

**Comment by seime** [comment]
I initally also spent some time evaluating JavaRule which is done according to JSR223, but I could not figure a way (back then) to build my rules in IntelliJ and deploy them as a compiled jar file. I neither got a clear picture of how to write unit tests.

To me a web editor has very little value. 

One thing I do miss is the ability to 

- See the rules and the triggers inside openHAB gui (just to get an overview)
- The ability to enable/disable rules quickly

I started to fiddle with this a few weeks back, but other priorities took over. Pushed the code at #90 but I only got so far that I could register a single hardcoded rule.

**Comment by querdenker2k** [comment]
I never used the feature to see all my rules in the main-ui. (The ui crashes the browser every time, since over a year)
Having this JSR223 compatible is nice, but i don't want to miss any current features of JRule.

**Comment by seaside1** [comment]
Nice to see that we sort of agree. PR 90 looks nice, yes would be nice to be able to enable / disable rules in the ui.

**Comment by querdenker2k** [comment]
Will there be no option to get JRule officially into openHAB because of that?

**Comment by seime** [comment]
IMHO I personally have no goal of trying to get it into the official distro. I've done it twice, and it is an extremely lengthy and time consuming task. I'd rather spend time improving the marketplace functionality to make unofficial releases easier and more visible to the user.

However, this is up @seaside1 to pursue if he wishes :)

**Comment by seaside1** [comment]
I'm not sure. Seems like rules languages in general are a bit sensitive in openHAB. I don't think they would want to merge unless we are JSR compliant. It might also be more difficult to maintain and do changes since the maintainers would take decisions on what to merge.

On the other hand JRule might mature and we might  feel like we don't have that much to add, or do big refactorings, then we might try.
